Formula/crispy-doom.rb

```diff
@@ -0,0 +1,43 @@
+class CrispyDoom < Formula
+  desc "Limit-removing enhanced-resolution Doom source port based on Chocolate Doom"
+  homepage "https://github.com/fabiangreffrath/crispy-doom"
+  url "https://github.com/fabiangreffrath/crispy-doom/archive/crispy-doom-5.10.0.tar.gz"
+  sha256 "0b6bac20816d057692e4f5c15dbbb27cc49e404e821f774e2b16ce997a6f3fe1"
+  license "GPL-2.0-only"
+
+  head do
+    url "https://github.com/fabiangreffrath/crispy-doom"
+  end
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libpng"
+  depends_on "libsamplerate"
+  depends_on "sdl2"
+  depends_on "sdl2_mixer"
+  depends_on "sdl2_net"
+
+  def install
+    system "autoreconf", "-fiv"
+    system "./configure", "--prefix=#{prefix}",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--disable-sdltest"
+    system "make", "install", "execgamesdir=#{bin}"
+  end
+
+  def caveats
+    <<~EOS
+      Note that this formula only installs a Doom game engine, and no
+      actual levels. The original Doom levels are still under copyright,
+      so you can copy them over and play them if you already own them.
+      Otherwise, there are tons of free levels available online.
+      Try starting here: https://www.chocolate-doom.org/
+    EOS
+  end
+
+  test do
+    assert_match "Crispy Doom #{version}", shell_output("#{bin}/crispy-doom -nogui", 255)
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.